components/fxa-client/src/oauth.rs

```diff
@@ -66,7 +68,12 @@ impl FirefoxAccount {
         let since_epoch = SystemTime::now()
             .duration_since(UNIX_EPOCH)
             .map_err(|_| ErrorKind::IllegalState("Current date before Unix Epoch."))?;
-        let expires_at = since_epoch.as_secs() + resp.expires_in;
+        let expires_in = if 0 < ttl && ttl < resp.expires_in {
```

Comment:
Can we push the "magic number" logic in `components/fxa-client/ffi/src/lib.rs` instead, and take a `ttl: Option<u64>` parameter in this function?

Reply:
Agreed, `Option<u64>` is the idiomatic rust way to do this, and I think that converting from "what's convenient for the FFI" to "what's idiomatic rust" is currently the job of the FFI bindings code.